odoo/orm/fields_misc.py

```diff
@@ -121,7 +122,7 @@ def __set__(self, record, value):
         raise TypeError("field 'id' cannot be assigned")
 
     def convert_to_column(self, value, record, values=None, validate=True):
-        return int(value)
```

Comment:
Did we consider checking `self.type` to do the casting for instance? Why do we have casting here anyway?
Or having different `Id` sub classes per type?

Overkill?

Reply:
Overkill. `int` is implicitly expected in many cases in the ORM. This PR just brings back the current usage of another type in a hand-written SQL view/query.